### PR TITLE
PROGRAMMES-6156 - Fix for network and service with same name

### PR DESCRIPTION
**Pull request by felipeparaujo**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/DsShared/Helpers/BroadcastNetworksHelper.php [hunk 1]
```diff
@@ -124,14 +133,27 @@ public function getCollapsedBroadcastNetworkBreakdown(CollapsedBroadcast $collap
                     ];
                 }
 
-                $breakdowns[$nid]['on_services'][(string) $service->getSid()] = $service;
+                // If the service has the same name as the network, ignore it
+                if (strtoupper($service->getShortName()) !== strtoupper($service->getNetwork()->getName())) {
```
Comment (votemike) [inline src/DsShared/Helpers/BroadcastNetworksHelper.php]: You should be able to use `strcasecmp()`, dunno if that's cleaner.

File: src/DsShared/Helpers/BroadcastNetworksHelper.php [hunk 2]
```diff
@@ -70,8 +71,10 @@ public function buildServicesNames(array $networkBreakdowns): array
             $broadcastOnServices = $networkBreakdown['on_services'];
             $notBroadcastOnServices = $networkBreakdown['not_on_services'];
 
-            if (empty($notBroadcastOnServices)) {
+            if (empty($notBroadcastOnServices) || empty($broadcastOnServices)) {
```
Comment (votemike) [inline src/DsShared/Helpers/BroadcastNetworksHelper.php]: When would `$broadcastOnServices` be empty?

Comment (felipeparaujo) [reply src/DsShared/Helpers/BroadcastNetworksHelper.php]: When the only service that got the broadcast has the same name as the network

Comment (votemike) [reply src/DsShared/Helpers/BroadcastNetworksHelper.php]: Our system is SO grim and convoluted 😿

**Pull request merged**